Add issue assignment management

ian-howell opened the pull request:
This commit allows the bot to assign issues to users using the `/assign`
command.

I went with `/assign` in place of the suggested `!assignMe` because the former echos what the kubernetes bot uses, which will be more familiar to open source contributors

closes #5

ian-howell committed:
Add issue assignment management

This commit allows the bot to assign issues to users using the `/assign`
command.

alexander-hughes commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+    # find the most recent assignment request
+    assignment_request = None
+    for comment in issue.get_comments().reversed:
+        if comment.body == '/assign':
```
should this be a contains '/assign' in case someone adds an unnecessary space or extra details like "Hi please /assign this to me."

ian-howell replied:
good call. updated

ian-howell committed:
Do substring check for /assign

This makes it so that if a user adds an extra space, or uses a more
involved message such as "Hi please /assign this to me", the bot will
still assign the user

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+
+
+def assign_issues(repo)
+    open_issues = [i for i in repo.get_issues() if i.state == 'open']
```
This makes me wonder if non-cores have the ability to reopen issues? Might be another command to add if not

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+        issue.create_comment(f'assigned {assignment_request.user.login}')
+        return
+
+    if issue_age(issue) > 60:
```
2 months seems a little long, but I'm not sure if an official "stale" timeline was established

ian-howell replied:
I don't think any specific timeline has ever been officially decided on for this kind of thing, but I do agree that 2 months is a bit long. I'm pretty sure I've seen other bots which mark an issue as "stale" when it reaches a month old, so I'll decrease this to 30 days. It can be changed if we notice it's a problem.

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+        for assignee in old_assignees:
+            issue.remove_from_assignees(assignee)
+        issue.add_to_assignees(assignment_request.user)
+        comment_body = f'unassigned: {", ".join([a for a in old_assinees])}\n' +
```
needs end of line \

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+    # If we've made it here, a user has requested to be assigned to a non-stale
+    # issue which is already assigned. Just notify the core team and let them
+    # handle the conflict.
+    comment_body = f'Unable to assign {assignment_request.user.login}. Please '+
```
needs end of line \ and space between ' and +

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+    # issue which is already assigned. Just notify the core team and let them
+    # handle the conflict.
+    comment_body = f'Unable to assign {assignment_request.user.login}. Please '+
+                   f'contact a member of the @airshipit/airship-cores team for '
```
needs end of line \

ian-howell committed:
Reduce "time until stale" from 2 months to 1

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+        for assignee in old_assignees:
+            issue.remove_from_assignees(assignee)
+        issue.add_to_assignees(assignment_request.user)
+        comment_body = f'unassigned: {", ".join([a for a in old_assinees])}\n' + \
```
old_assinees should be old_assignees

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+            issue.remove_from_assignees(assignee)
+        issue.add_to_assignees(assignment_request.user)
+        comment_body = f'unassigned: {", ".join([a for a in old_assinees])}\n' + \
+                       f'assigned {assignment_request.user.login}'
```
nit: ":" after assigned

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+    # handle the conflict.
+    comment_body = f'Unable to assign {assignment_request.user.login}. Please ' + \
+                   f'contact a member of the @airshipit/airship-cores team for ' + \
+                   f'help with assignments'
```
nit: punctuation

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
             return i
+
+
+def assign_issues(repo)
```
missing colon

ianpittwood commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +75,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+
+
+def issue_age(issue):
+    return (datetime.now() - issue.created_at).days
```
you'll need to import datetime

ianpittwood reviewed (changes requested):
Please use type hints for the new functions too to fit in better with our current code style

ian-howell committed:
Fixup bugs for issue assignment

* Added missing colon
* Added missing `datetime` import
* Fixed a misspelling

alexander-hughes commented on gerrit_to_github_issues/github_issues.py:
```diff
@@ -75,3 +76,52 @@ def get_bot_comment(issue: Issue, bot_name: str, ps_number: str) -> IssueComment
+        issue.create_comment(f'assigned {assignment_request.user.login}')
+        return
+
+    if issue_age(issue) > 30:
+        # If the issue is 2 months old and the original assignees haven't
```
age is only being compared to 1 month but the comment suggests 2 months.

ian-howell replied:
fixed
